client/blocks/inline-help/inline-help-search-results.jsx

```diff
@@ -123,13 +130,32 @@ function HelpSearchResults( {
 	};
 
 	const renderHelpLink = ( result ) => {
-		const { link, key, title, support_type = SUPPORT_TYPE_API_HELP, icon = 'domains' } = result;
+		const {
+			link,
+			key,
+			title,
+			support_type = SUPPORT_TYPE_API_HELP,
+			icon = 'domains',
+			post_id,
+		} = result;
 		const resultIndex = searchResults.findIndex( ( r ) => r.link === link );
 
 		const classes = classNames( 'inline-help__results-item', {
 			'is-selected': selectedResultIndex === resultIndex,
 		} );
 
+		// Unless searching with Inline Help or on the Purchases section, hide the
+		// "Managing Purchases" documentation link for users who have not made a purchase.
+		if (
+			post_id === 111349 &&
+			! isSearching &&
+			! hasAPIResults &&
+			! hasPurchases &&
+			sectionName !== 'purchases'
```

Comment:
Hi! I can't remember for certain, but I'm pretty sure that the intention was that people who were actively searching for things to do with purchasing should still be able to see the "Managing Purchases" documentation even if they haven't made a purchase.

Reply:
I think I understand the intent now: we want to hide the "Manage Purchases" article when it's part of the contextual results, like when in the `plans` section, but don't want to hide when it's a part of actual search results from the server. When I'm searching for "manage purchases", this article is one of the results returned by the server. We want to always show that.

Is that right?

Reply:
Yes, I believe that's correct.

Reply:
In that case, it could be implemented by filtering only the "contextual results" array, probably in the related selector, and leaving the "search results" array intact. @flootr that should be easy to implement now when they are in fact separate arrays.